### PR TITLE
Parameterize maximum number of tests pods to run in parallel

### PR DESCRIPTION
**Pull request by tingwai**
**What this PR does / why we need it**:

Currently the max number of test pods that can be run in parallel is hardcoded to 20. This PR allows that to be configurable with the `--max` flag. 

Example: `helm test release_name --parallel --max 99`

**Special notes for your reviewer**:

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by typokign — APPROVED** [review]
Make it so

File: pkg/releasetesting/environment.go
Comment (bacongobbler) [inline pkg/releasetesting/environment.go]: is there a way we can revert this? This would be considered a backwards compatibility breaker.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/releasetesting/environment.go
Comment (tingwai) [reply pkg/releasetesting/environment.go]: Good point, I'll revert back to uint32

Comment (tingwai) [reply pkg/releasetesting/environment.go]: Its back to uint32 now along with the other variables

**Comment by tingwai** [comment]
So the unit tests are consistently passing locally but fail in Circle, should I keep restarting until it works?

**Comment by hickeyma** [comment]
@tingwai Seems to be performance issues with Circleci at moment. I have just kicked off another build so lets see.

**Comment by tingwai** [comment]
@hickeyma Thanks, build is green now. Can I get a re-review?

**Comment by tingwai** [comment]
@bacongobbler @hickeyma Can I get a re-review for this PR? This PR is getting stale

cc'ing other maintainers to get this more exposure
@adamreese @michelleN

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @tingwai.

File: pkg/tiller/release_testing.go
Comment (bacongobbler) [inline pkg/tiller/release_testing.go]: this can be simplified:

```
parallelism := maxmaxParallelism
if req.MaxParallel == 0 {
	parallelism = maxParallelism
}
```

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/tiller/release_testing.go
Comment (tingwai) [reply pkg/tiller/release_testing.go]: Fixed. I had to do `parallelism := uint32(maxParallelism)` but same effect

**Comment by driscollAtBitGo** [comment]
@tingwai please don't let this go stale!

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
Because this is a `size/L`, this PR will need another core maintainer to review this before it can be merged. Thanks!

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @tingwai

**Comment by hickeyma** [comment]
@bacongobbler I am not sure that this PR can be ported to v3 because of the changes to the test infrastructure to hooks. What do you think?

**Pull request merged**
